pkg/apis/eventing/v1alpha1/trigger_defaults.go

```diff
@@ -16,10 +16,34 @@ limitations under the License.
 
 package v1alpha1
 
-import "context"
+import (
+	"context"
+	"github.com/knative/eventing/pkg/apis/eventing"
+	"github.com/knative/pkg/apis"
+	"k8s.io/apimachinery/pkg/api/equality"
+)
 
 func (t *Trigger) SetDefaults(ctx context.Context) {
 	t.Spec.SetDefaults(ctx)
+
+	if ui := apis.GetUserInfo(ctx); ui != nil {
```

Comment:
yeah, ducktype this and you can remove the 4 impls.

Reply:
Thanks! I will fix it later.